README.md

```diff
@@ -331,3 +331,44 @@ that can be tested with `./scripts/test/test_plugin.sh`.
 [processTargetModule]: liquidhaskell-boot/src/Language/Haskell/Liquid/GHC/Interface.hs#L483
 [processModule]:       liquidhaskell-boot/src/Language/Haskell/Liquid/GHC/Plugin.hs#L509
 
+# Module Verification Process
+
+The following graph summarizes the verification process of a module by showing
+its specification life cycle.
+See the [Specs module](./liquidhaskell-boot/src/Language/Haskell/Liquid/Types/Specs.hs)
+for detailed documentation on the specification stages.
+
+```mermaid
+flowchart LR
+  subgraph Liquid Haskell
+
+    subgraph Inputs
+      direction TB
+      H[
+        Dependencies of A
+        LifttedSpecs
+        ]
+      C[CoreBinds]
+      B[BarseSpec]
+    end
+
+    subgraph Outputs
+      direction TB
+      T[TargetSpec]
```

Comment:
If `liquid-fixpoint` is to stay in the diagram, the output of LF would be the verification contraints rather than the TargetSpec.
```suggestion
      T[Verification
Constraints]
```
This is spreading a bit beyond `Specs.hs`. `Liquid.hs` has the constraint generation part, and `SpecFinder.hs` has the part about finding dependencies and LHAssumptions.

Alternatively, both TargetSpec and verification constraints could be left in the diagram.